SyntaxErrorException when filtering certain dates

I received the following error when filtering by Approval Date, Committee Approval Date, Committee Embargo Approval Date, License Agreement Date, and Submission Date. The date shows under Now Filtering By, but the list is not filtered.

Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "PFV8.VALUE" not found; SQL statement:

<img width="1136" alt="Screenshot 2023-04-12 at 8 21 02 AM" src="https://user-images.githubusercontent.com/5767401/231516149-bb16640d-2110-4ed0-9ac9-50d04f2b6b20.png">

--- a/src/main/java/org/tdl/vireo/model/repo/impl/SubmissionRepoImpl.java
+++ b/src/main/java/org/tdl/vireo/model/repo/impl/SubmissionRepoImpl.java
@@ -737,9 +737,7 @@ public class SubmissionRepoImpl extends AbstractWeaverRepoImpl<Submission, Submi
 
                     for (String filterString : submissionListColumn.getFilters()) {
                         filterString.replaceAll("[TZ:.\\-]", " ");
-
-                        sqlWhereBuilderList.add(new StringBuilder()
-                            .append("CAST(s.submission_date AS TIMESTAMP) = '").append(filterString).append("'"));
+                        sqlWhereBuilderList.add(buildSubmissionDateFieldString("submission_date", filterString));
                     }
 
                     break;
@@ -751,9 +749,7 @@ public class SubmissionRepoImpl extends AbstractWeaverRepoImpl<Submission, Submi
 
                     for (String filterString : submissionListColumn.getFilters()) {
                         filterString.replaceAll("[TZ:.\\-]", " ");
-
-                        sqlWhereBuilderList.add(new StringBuilder()
-                            .append("CAST(s.approve_application_date AS TIMESTAMP) = '").append(filterString).append("'"));
+                        sqlWhereBuilderList.add(buildSubmissionDateFieldString("approve_application_date", filterString));
                     }
 
                     break;
@@ -765,9 +761,7 @@ public class SubmissionRepoImpl extends AbstractWeaverRepoImpl<Submission, Submi
 
                     for (String filterString : submissionListColumn.getFilters()) {
                         filterString.replaceAll("[TZ:.\\-]", " ");
-
-                        sqlWhereBuilderList.add(new StringBuilder()
-                            .append("CAST(s.approve_advisor_date AS TIMESTAMP) = '").append(filterString).append("'"));
+                        sqlWhereBuilderList.add(buildSubmissionDateFieldString("approve_advisor_date", filterString));
                     }
 
                     break;
@@ -779,9 +773,7 @@ public class SubmissionRepoImpl extends AbstractWeaverRepoImpl<Submission, Submi
 
                     for (String filterString : submissionListColumn.getFilters()) {
                         filterString.replaceAll("[TZ:.\\-]", " ");
-
-                        sqlWhereBuilderList.add(new StringBuilder()
-                            .append("CAST(s.approve_embargo_date AS TIMESTAMP) = '").append(filterString).append("'"));
+                        sqlWhereBuilderList.add(buildSubmissionDateFieldString("approve_embargo_date", filterString));
                     }
 
                     break;
@@ -891,6 +883,21 @@ public class SubmissionRepoImpl extends AbstractWeaverRepoImpl<Submission, Submi
         default:
             break;
         }
+    }
+
+    /**
+     * Build a submission date field string given some filter.
+     *
+     * This is form submission date fields that are already stored in the SQL date format.
+     *
+     * @param column The column name to filter.
+     * @param filter The filter.
+     * @return A constructed string builder appropriately casting the date.
+     */
+    private StringBuilder buildSubmissionDateFieldString(String column, String filter) {
+        return new StringBuilder()
+            .append("CAST(s.").append(column)
+            .append(" AS TIMESTAMP) = '").append(filter).append("'");
     }
 
     @Override

